src/rest.cpp

```diff
@@ -607,7 +607,7 @@ static bool rest_blockhash_by_height(HTTPRequest* req,
     std::string height_str;
     const RetFormat rf = ParseDataFormat(height_str, str_uri_part);
 
-    int32_t blockheight;
+    int32_t blockheight = 0;
```

Comment:
```suggestion
    int32_t blockheight = 0; // Initialization only needed as workaround for -O2 compiler optimization and valgrind interaction, see https://github.com/bitcoin/bitcoin/pull/18785
```

This will prevent valgrind from detecting real issues, so maybe add a comment with motivation?

Reply:
re: https://github.com/bitcoin/bitcoin/pull/18785/files#r416090195

> This will prevent valgrind from detecting real issues, so maybe add a comment with motivation?

Thanks, added comment. I'm not sure if it is too likely to be helpful to someone reading the code, but I guess it's better to have an explanation for something you wouldn't expect to be necessary.

I also changed init value from 0 to -1 just to make it clear that if height is not parsed it will always trigger the error, regardless of the bool return value, and there's no strange case where this is supposed to return information about block 0